pkg/build/util/util.go

```diff
@@ -307,3 +307,41 @@ func SafeForLoggingS2IConfig(config *s2iapi.Config) *s2iapi.Config {
 	newConfig.ScriptsURL, _ = s2iutil.SafeForLoggingURL(newConfig.ScriptsURL)
 	return &newConfig
 }
+
+// GetBuildEnv gets the build strategy environment
+func GetBuildEnv(build *buildapi.Build) *[]kapi.EnvVar {
+	switch {
+	case build.Spec.Strategy.SourceStrategy != nil:
+		return &build.Spec.Strategy.SourceStrategy.Env
+	case build.Spec.Strategy.DockerStrategy != nil:
+		return &build.Spec.Strategy.DockerStrategy.Env
+	case build.Spec.Strategy.CustomStrategy != nil:
+		return &build.Spec.Strategy.CustomStrategy.Env
+	case build.Spec.Strategy.JenkinsPipelineStrategy != nil:
+		return &build.Spec.Strategy.JenkinsPipelineStrategy.Env
+	default:
+		return nil
+	}
+}
+
+// UpdateBuildEnv updates the strategy environment
+// This will replace the existing variable definitions with provided env
+func UpdateBuildEnv(build *buildapi.Build, env []kapi.EnvVar) {
```

Comment:
I don't actually think we want to disable ValueFrom in buildrequests.  Why would it be a problem if it does?  We'd end up with a build object that contains ValueFrom envs, and then when the build gets accepted, the ValueFroms will be resolved just as if the original BuildConfig had ValueFrom refs, right?

(probably also worth noting this function is just slightly refactored from what was already there.  i'm not even entirely sure why it was made public since there are no new callers being introduced)